R/DataLoadingSaving.R

```diff
@@ -360,45 +351,135 @@ getDbCohortMethodData <- function(connectionDetails,
   )
-  outcomes <- DatabaseConnector::querySql(connection, outcomeSql, snakeCaseToCamelCase = TRUE)
+  outcomes <-
+    DatabaseConnector::querySql(connection, outcomeSql, snakeCaseToCamelCase = TRUE)
   outcomes$rowId <- as.numeric(outcomes$rowId)
   metaData$outcomeIds <- outcomeIds
   delta <- Sys.time() - start
-  message("Fetching outcomes took ", signif(delta, 3), " ", attr(delta, "units"))
+  message("Fetching outcomes took ",
+          signif(delta, 3),
+          " ",
+          attr(delta, "units"))
   ParallelLogger::logDebug("Fetched outcomes total count is ", nrow(outcomes))
 
   # Remove temp tables:
-  renderedSql <- SqlRender::loadRenderTranslateSql("RemoveCohortTempTables.sql",
-    packageName = "CohortMethod",
-    dbms = connectionDetails$dbms,
-    tempEmulationSchema = tempEmulationSchema,
-    sampled = sampled
-  )
+  renderedSql <-
+    SqlRender::loadRenderTranslateSql(
+      "RemoveCohortTempTables.sql",
+      packageName = "CohortMethod",
+      dbms = connectionDetails$dbms,
+      tempEmulationSchema = tempEmulationSchema,
+      sampled = sampled
+    )
   DatabaseConnector::executeSql(connection,
-    renderedSql,
-    progressBar = FALSE,
-    reportOverallTime = FALSE
-  )
+                                renderedSql,
+                                progressBar = FALSE,
+                                reportOverallTime = FALSE)
 
   covariateData$cohorts <- cohorts
   covariateData$outcomes <- outcomes
-  attr(covariateData, "metaData") <- append(attr(covariateData, "metaData"), metaData)
+  attr(covariateData, "metaData") <-
+    append(attr(covariateData, "metaData"), metaData)
   class(covariateData) <- "CohortMethodData"
   attr(class(covariateData), "package") <- "CohortMethod"
   return(covariateData)
 }
 
-handleCohortCovariateBuilders <- function(covariateSettings, exposureDatabaseSchema, exposureTable) {
-  if (is(covariateSettings, "covariateSettings")) {
-    covariateSettings <- list(covariateSettings)
-  }
-  for (i in 1:length(covariateSettings)) {
-    object <- covariateSettings[[i]]
-    if ("covariateCohorts" %in% names(object) && is.null(object$covariateCohortTable)) {
-      object$covariateCohortDatabaseSchema <- exposureDatabaseSchema
-      object$covariateCohortTable <- exposureTable
-      covariateSettings[[i]] <- object
+downSample <-
+  function(connection,
+           tempEmulationSchema,
+           targetId,
+           maxCohortSize,
+           sampled) {
+    renderedSql <- SqlRender::loadRenderTranslateSql(
+      "CountCohorts.sql",
+      packageName = "CohortMethod",
+      dbms = connection@dbms,
+      tempEmulationSchema = tempEmulationSchema,
+      target_id = targetId
+    )
+    counts <-
+      DatabaseConnector::querySql(connection, renderedSql, snakeCaseToCamelCase = TRUE)
+    ParallelLogger::logDebug("Pre-sample total row count is ", sum(counts$rowCount))
+    preSampleCounts <- dplyr::tibble(dummy = 0)
+    idx <- which(counts$treatment == 1)
```

Comment:
Why not move the computing of `idx` to the `preSample()` function? You can pass which treatment (0 or 1) as an argument, which would allow you to remove the `colType` argument.

Reply:
Done: 3056327b83780d34bdf939e81badb7533155593e.

However, the implementation is a bit more verbose, let me know what you prefer.